diff --git a/node_modules/react-native-sound/sound.js b/node_modules/react-native-sound/sound.js
index c9bcbaf..0bed368 100644
--- a/node_modules/react-native-sound/sound.js
+++ b/node_modules/react-native-sound/sound.js
@@ -4,7 +4,10 @@ var ReactNative = require('react-native');
 var RNSound = ReactNative.NativeModules.RNSound;
 var IsAndroid = RNSound.IsAndroid;
 var IsWindows = RNSound.IsWindows;
-var resolveAssetSource = require("react-native/Libraries/Image/resolveAssetSource");
+var resolveAssetSourceModule = require("react-native/Libraries/Image/resolveAssetSource");
+var resolveAssetSource = resolveAssetSourceModule && resolveAssetSourceModule.__esModule
+  ? resolveAssetSourceModule.default
+  : resolveAssetSourceModule;
 var eventEmitter = new ReactNative.NativeEventEmitter(RNSound);
 
 var nextKey = 0;
