--- ggml.c.orig	2025-09-14 13:51:37
+++ ggml.c	2025-09-14 13:40:23
@@ -1,5 +1,13 @@
 #define _CRT_SECURE_NO_DEPRECATE // Disables "unsafe" warnings on Windows
 #define _USE_MATH_DEFINES // For M_PI on MSVC
+
+// GGML build info
+#ifndef GGML_VERSION
+#define GGML_VERSION "unknown"
+#endif
+#ifndef GGML_COMMIT
+#define GGML_COMMIT "unknown"
+#endif

 #include "ggml-backend.h"
 #include "ggml-impl.h"
