--- ggml.c.orig	2024-12-20 13:38:33
+++ ggml.c	2024-12-20 13:38:34
@@ -118,9 +118,9 @@ static void wsp_ggml_print_backtrace_symbols(void) {
 #elif defined(__linux__) && defined(__GLIBC__)
 #include <execinfo.h>
 static void wsp_ggml_print_backtrace_symbols(void) {
-    void * trace[100];
-    int nptrs = backtrace(trace, sizeof(trace)/sizeof(trace[0]));
-    backtrace_symbols_fd(trace, nptrs, STDERR_FILENO);
+    // void * trace[100];
+    // int nptrs = backtrace(trace, sizeof(trace)/sizeof(trace[0]));
+    // backtrace_symbols_fd(trace, nptrs, STDERR_FILENO);
 }
 #else
 static void wsp_ggml_print_backtrace_symbols(void) {
