--- ggml-metal.m.orig	2024-11-19 13:30:22
+++ ggml-metal.m	2024-11-19 13:30:23
@@ -461,7 +461,7 @@
         const bool try_metallib = true;
 #endif

-        NSString * path_lib = [bundle pathForResource:@"default" ofType:@"metallib"];
+        NSString * path_lib = [bundle pathForResource:@"ggml-whisper" ofType:@"metallib"];
         if (try_metallib && path_lib != nil) {
             // pre-compiled library found
             NSURL * libURL = [NSURL fileURLWithPath:path_lib];
