--- ggml-backend-reg.cpp.orig	2024-11-17 11:53:44
+++ ggml-backend-reg.cpp	2024-11-17 11:53:17
@@ -12,9 +12,14 @@
 #endif

 #ifdef WSP_GGML_USE_METAL
+#include <TargetConditionals.h>
+
+#if !TARGET_OS_SIMULATOR
 #include "ggml-metal.h"
 #endif

+#endif
+
 #ifdef WSP_GGML_USE_SYCL
 #include "ggml-sycl.h"
 #endif
@@ -52,8 +57,12 @@
         register_backend(wsp_ggml_backend_cuda_reg());
 #endif
 #ifdef WSP_GGML_USE_METAL
+
+#if !TARGET_OS_SIMULATOR
         register_backend(wsp_ggml_backend_metal_reg());
 #endif
+
+#endif
 #ifdef WSP_GGML_USE_SYCL
         register_backend(wsp_ggml_backend_sycl_reg());
 #endif
