--- whisper.cpp.orig	2024-11-03 12:39:40
+++ whisper.cpp	2024-11-03 12:40:25
@@ -3388,8 +3388,10 @@
         const size_t memory_size = aheads_masks_nbytes(state->aheads_masks);
         WHISPER_LOG_INFO("%s: alignment heads masks size = %ld B\n", __func__, memory_size);
     }
+

 #ifdef WHISPER_USE_COREML
+    if (ctx->params.use_coreml) {
     const auto path_coreml = whisper_get_coreml_path_encoder(ctx->path_model);

     WHISPER_LOG_INFO("%s: loading Core ML model from '%s'\n", __func__, path_coreml.c_str());
@@ -3404,6 +3406,7 @@
 #endif
     } else {
         WHISPER_LOG_INFO("%s: Core ML model loaded\n", __func__);
+    }
     }
 #endif

@@ -3558,6 +3561,7 @@
 struct whisper_context_params whisper_context_default_params() {
     struct whisper_context_params result = {
         /*.use_gpu              =*/ true,
+        /*.use_coreml           =*/ false,
         /*.flash_attn           =*/ false,
         /*.gpu_device           =*/ 0,

