--- whisper.h.orig	2024-11-07 14:41:36
+++ whisper.h	2024-11-07 14:37:28
@@ -114,6 +114,7 @@

     struct whisper_context_params {
         bool  use_gpu;
+        bool  use_coreml;
         bool  flash_attn;
         int   gpu_device;  // CUDA device

@@ -423,6 +424,20 @@
     WHISPER_API whisper_token whisper_token_transcribe(struct whisper_context * ctx);

     // Performance information from the default state.
+    struct whisper_timings {
+        int64_t load_us;
+        int64_t t_start_us;
+        int32_t fail_p;
+        int32_t fail_h;
+        float mel_ms;
+        float sample_ms;
+        float encode_ms;
+        float decode_ms;
+        float batchd_ms;
+        float prompt_ms;
+    };
+
+    WHISPER_API struct whisper_timings whisper_get_timings(struct whisper_context * ctx);
     WHISPER_API void whisper_print_timings(struct whisper_context * ctx);
     WHISPER_API void whisper_reset_timings(struct whisper_context * ctx);

